src/server/sample_data/routes/lib/load_data.js

```diff
@@ -0,0 +1,78 @@
+import readline from 'readline';
+import fs from 'fs';
+import zlib from 'zlib';
+
+const BULK_LOAD_SIZE = 500;
+
+export function loadData(path, bulkLoad, callback) {
+  let count = 0;
+  let docs = [];
+  let isPaused = false;
+
+  const readStream = fs.createReadStream(path, {
+    // pause does not stop lines already in buffer. Use smaller buffer size to avoid bulk inserting to many records
+    highWaterMark: 1024 * 4
+  });
+  const lineStream = readline.createInterface({
+    input: readStream.pipe(zlib.Unzip()) // eslint-disable-line new-cap
+  });
+
+  const onClose = async () => {
+    if (docs.length > 0) {
+      try {
+        await bulkLoad(docs);
+      } catch (err) {
+        callback(err);
+        return;
+      }
+    }
+    callback(null, count);
+  };
+  lineStream.on('close', onClose);
+
+  function closeWithError(err) {
+    lineStream.removeListener('close', onClose);
+    lineStream.close();
+    callback(err);
+  }
+
+  lineStream.on('line', async (line) => {
+    if (line.length === 0 || line.charAt(0) === '#') {
+      return;
+    }
+
+    let doc;
+    try {
+      doc = JSON.parse(line);
```

Comment:
This means we can't have newlines within our JSON, objects, eh? If we're going to expose sample data in such a way that that other plugin authors can call `registerSampleDataSet`, then this constraint should be added to the docs on this feature (if / when such docs are written), as I'd find it surprising.